rest-proxy-core/src/main/groovy/edu/wisc/my/restproxy/dao/AgnosticHttpMessageConverter.groovy

```diff
@@ -0,0 +1,53 @@
+package edu.wisc.my.restproxy.dao
+
+import com.google.common.io.ByteStreams
+import org.apache.commons.lang3.NotImplementedException
+import org.springframework.http.HttpInputMessage
+import org.springframework.http.HttpOutputMessage
+import org.springframework.http.MediaType
+import org.springframework.http.converter.AbstractHttpMessageConverter
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody
+
+/**
+ * This converter supports reading all content types into an {@link Object} by copying the byte stream.
+ * It does not write any requests.
+ *
+ * @author Craig Knuth
+ */
+class AgnosticHttpMessageConverter extends AbstractHttpMessageConverter<Object> {
```

Comment:
Personally I like "AgnosticHttpMessageConverter". I think it's a little confusing to have a class called HttpMessageConverter that extends AbstractHttpMessageConverter that implements HttpMessageConverter. If the class name lacked brevity then I would consider it, but this is fine in terms of length.